### DISCUSSION TITLE
Issues upgrading to Percy Storybook v4

### DISCUSSION THREAD

**Discussion by jeffchew** [percy/percy-storybook#566] [Support]
## The problem

We have been having issues upgrading `@percy/storybook` from v3 to v4. We have made the suggestion of removing any ad service calls (by removing them entirely) but still seems to be an issue. The end result is that most snapshots go missing. This is observed in both our React components and Web Components versions of Storybook:

**PR opened:**
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/8128

**React Percy latest result:**
https://percy.io/538fc19a/Carbon-for-IBM.com-React/builds/18147514?utm_campaign=538fc19a&utm_content=Carbon-for-IBM.com-React&utm_source=github_status_public

<img width="228" alt="image" src="https://user-images.githubusercontent.com/1641214/168880462-16317ae9-0841-42ed-8d6b-8d9b687bd6a1.png">

**Web Components Percy latest result:**
https://percy.io/538fc19a/Carbon-for-IBM.com-Web-Components/builds/18148038?utm_campaign=538fc19a&utm_content=Carbon-for-IBM.com-Web-Components&utm_source=github_status_public

<img width="235" alt="image" src="https://user-images.githubusercontent.com/1641214/168880617-f2a45060-25c9-4f18-995d-82d71b5fc71c.png">

## Environment

- Node version: `14`
- `@percy/cli` version: `1.2.0`
- Version of Percy SDK you’re using: `4.2.0`
- If needed, a build or snapshot ID: see above
- OS version: Ubuntu 20
- Type of shell command-line [interface]: Linux (Jenkins on Kubernetes)
- Storybook v5 (we are in the process of migrating to v6)

## Details

When digging into the logs, it appears that JS artifacts themselves (produced by Storybook) are getting blocked, an example of what we're seeing:

<img width="1615" alt="image" src="https://user-images.githubusercontent.com/1641214/168880756-e5fadc5c-b152-4c18-a5f9-16cf15f70c9d.png">

<img width="1594" alt="image" src="https://user-images.githubusercontent.com/1641214/168880846-f796d8cf-0a2d-4338-add9-bc62ec5a4802.png">

## Debug logs

Attached:
[Deploy Preview build log 20220517.txt](https://github.com/percy/percy-storybook/files/8710589/Deploy.Preview.build.log.20220517.txt)

## Code to reproduce issue

You can probably best see using the fork:
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/8128

Steps to build:
1) `yarn install`
2) `yarn build`
3) `cd packages/react`
4) `yarn build-storybook`
5) `yarn visual-snapshot`

**Comment by Robdel12** [comment]
Hey @jeffchew! I'd give 4.3.0 a try: https://github.com/percy/percy-storybook/releases/tag/v4.3.0 we cleaned up some things which should hopefully clear up these errors.

**Comment by jeffchew** [comment]
Thank you @Robdel12 ! Gave it a try, and looks like the React storybook went through ok but still seeing problems with web components. A new PR was opened since the previous one was way out of sync with `main` branch, so you can see results in this PR:

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/9068

The percy runs:

https://percy.io/538fc19a/Carbon-for-IBM.com-React/builds/19479186?utm_campaign=538fc19a&utm_content=Carbon-for-IBM.com-React&utm_source=github_status_public
https://percy.io/538fc19a/Carbon-for-IBM.com-Web-Components/builds/19479416/changed/1094041757?browser=chrome&browser_ids=23&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280

Attached logs for the web components run:
[Carbon for IBM dotcom web components percy 2022-07-08.txt](https://github.com/percy/percy-storybook/files/9075011/Carbon.for.IBM.dotcom.web.components.percy.2022-07-08.txt)

The following bumps were made for that particular package in our monorepo:
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/9068/files#diff-7298512f791c54d2827facfc8d26ae8a5db2ac960e464e30953ccaab05dd412eR146-R150
